api/api.py

```diff
@@ -100,33 +101,50 @@ def host_response():
         target = request.json['target']
         if action == 'blacklist':
-            utils.flush_ipset_list('blacklist-user')
-            for ip in target:
-                with lock:
-                    if (len(ip) >= 7) and (ip not in homenet.user_blacklist) and (ip not in homenet.user_whitelist):
-                        homenet.user_blacklist.append(ip)
-                        utils.add_ip_ipset_blacklist(ip, 'blacklist-user')
-                        log.debug('FG-INFO: IP ' + ip + 'added to user blacklist')
+            if re.search('[a-zA-Z]', target):
+                for domain in target:
+                    with lock:
+                        if (domain not in homenet.user_whitelist) and (domain not in homenet.user_blacklist):
+                            homenet.user_blacklist.append(domain)
+                            log.debug('FG-INFO: Domain ' + domain + ' added to user blacklist')
+            else:
+                utils.flush_ipset_list('blacklist-user')
+                for ip in target:
+                    with lock:
+                        if (len(ip) >= 7) and (ip not in homenet.user_blacklist) and (ip not in homenet.user_whitelist):
+                            homenet.user_blacklist.append(ip)
+                            utils.add_ip_ipset_blacklist(ip, 'blacklist-user')
+                            log.debug('FG-INFO: IP ' + ip + 'added to user blacklist')
 
             resp = Response()
             resp.status_code = 200
             return resp
         elif action == 'unblock':
-            for ip in target:
-                with lock:
-                    if (len(ip) >= 7) and (ip in homenet.user_blacklist):
-                        utils.del_ip_ipset_blacklist(ip, 'blacklist-user')
-            resp = Response()
-            resp.status_code = 200
-            return resp
+            if re.search('[a-zA-Z]', target):
+                for domain in target:
+                    utils.del_domain_blacklist(domain)
+            else:
+                for ip in target:
+                    with lock:
+                        if (len(ip) >= 7) and (ip in homenet.user_blacklist):
+                            utils.del_ip_ipset_blacklist(ip, 'blacklist-user')
+                    resp = Response()
+                    resp.status_code = 200
+                    return resp
         elif action == 'whitelist':
-            for ip in target:
-                if len(ip) >= 7:
-                    utils.del_ip_ipset_blacklist(ip, 'blacklist')
-                    utils.del_ip_ipset_blacklist(ip, 'blacklist-user')
+            if re.search('[a-zA-Z]', target):
+                for domain in target:
                     with lock:
-                        if ip not in homenet.user_whitelist:
-                            homenet.user_whitelist.append(ip)
+                        if domain not in homenet.user_whitelist:
+                            homenet.user_whitelist.append(domain)
```

Comment:
There's a similar issue here like "user_blacklist". The "user_whitelist" list was used before to store only IP addresses so using it now for mixing IPs and domains might bring undesired issues somewhere else in the program. As before, I suggest you to create and use a dedicated list instead in the Network class located in objects.py.